flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveRecordSerDe.java

```diff
@@ -47,6 +56,12 @@ public static Object obtainFlinkRowField(Object field, ObjectInspector fieldObje
 		Object res;
 		if (fieldObjectInspector.getCategory() == ObjectInspector.Category.PRIMITIVE) {
 			res = convertPrimitiveField(field, (PrimitiveObjectInspector) fieldObjectInspector);
+		} else if (fieldObjectInspector.getCategory() == ObjectInspector.Category.STRUCT) {
```

Comment:
maybe we can just use `HiveInspector.toFlinkObject()` and remove `obtainFlinkRowField()`?

Reply:
You are right. I have removed this class in the latest commit.